develop/reference/articles/06-themes/02-theme-components-and-workflow.markdown

```diff
@@ -0,0 +1,238 @@
+# Theme Components and Workflow [](id=theme-components-and-workflow)
+
+If you want to develop a website, you must have three key components: CSS, JS, 
+and HTML. @product@ supports SASS as well as multiple JavaScript frameworks. The 
+HTML, however, is rendered via 
+[FreeMarker](https://freemarker.apache.org/) 
+theme templates. This reference guide provides an overview of @product@'s theme 
+development components and workflow, covering the following topics:
+
+- Theme templates
+- Theme customizations and extensions
+- Portlet customizations and extensions
+- Theme workflow
+- CSS Frameworks and extensions
+
+## Theme Templates [](id=theme-templates)
+
+@product@ provides several default FreeMarker templates that each handle a key 
+piece of functionality for the page:
+
+- `portal_normal.ftl`: Similar to a static site’s `index.html`, this file acts 
+  as a hub for all theme templates, as well as provides the basic markup for the 
+  page.
+- `init.ftl`: Contains common FreeMarker variables that are available to use in 
+  your theme templates. Useful for reference if you need access to theme 
+  objects. **We recommended that you DO NOT overwrite this file**.
+- `init_custom.ftl`: Used to overwrite FreeMarker variables found in `init.ftl` 
+  and to define any new ones such as 
+  [theme setting](/develop/tutorials/-/knowledge_base/7-1/making-configurable-theme-settings) 
+  variables.
+- `portlet.ftl`: The theme template for the theme’s portlets. If your theme uses 
+  Portlet Decorators, you can modify this file to create application 
+  decorator-specific theme settings. See the 
+  [Portlet Decorators](/develop/tutorials/-/knowledge_base/7-1/creating-configurable-styles-for-portlet-wrappers) 
+  tutorial for more info.
+- `navigation.ftl`: Contains the navigation markup. Note that this 
+  implementation requires some modifying to work properly in 7.1, as the 
```

Comment:
Features have been removed. That means, it works as intended. Saying that this requires something to work properly is either confusing or unnecessary. If this is not what should be used, maybe we shouldn't show it?

Reply:
I provided the information based on @pat270 's description. He said that the pattern was unusable in 7.1 and it has to be modified with the navigation macro. That sounded like it was broken to me. I can reword though to say "make sure you use the navigation macro in your template." or something like that so it doesn't sound so negative, since it does still work. It just requires the macro.

Reply:
@jbalsas here is the proposed update:

> - `navigation.ftl`: Contains the navigation markup. To customize pages in the navigation, you must use the `liferay.navigation_menu` macro. This lets you leverage [ADTs](https://github.com/liferay/liferay-portal/tree/7.1.x/modules/apps/site-navigation/site-navigation-menu-web/src/main/resources/com/liferay/site/navigation/menu/web/portlet/template/dependencies) for the navigation menu.

Reply:
LGTM!